### PR TITLE
Add options to menu

### PR DESCRIPTION
**Pull request by gjmooney**
*Issue number of the reported bug or feature request: #<number>*

**Describe your changes**
Added several option to the column context menu
* Option to clear sort selection
* Option to copy selection to the clipboard
* Option to download selection as a CSV
* Option to download the entire table as a CSV
* Option to clear current selection
* Added separators between sections of the menu

**Testing performed**
Since this is mostly just adding UI elements for existing capabilities only manual testing was performed in addition to running to the current test suite

**Additional context**
Selection based options are greyed out when there is no current selection

### CODE REVIEW

File: js/feathergrid.ts [hunk 1]
```diff
@@ -434,10 +433,6 @@ export class FeatherGrid extends Widget {
       commands: this._createCommandRegistry(),
     });
 
-    // Replace method of copying to clipboard with one that allows
-    // a ClipboardEvent to reach document.body
-    this.grid.copyToClipboard = this.copyToClipboard.bind(this.grid);
```
Comment (martinRenou) [inline js/feathergrid.ts:439]: some welcome cleaning 🧹, thanks!

File: js/feathergrid.ts [hunk 2]
```diff
@@ -994,6 +989,56 @@ export class FeatherGrid extends Widget {
         },
       },
     );
+    commands.addCommand(
+      FeatherGridContextMenu.CommandID.CopySelectionToClipboard,
+      {
+        label: 'Copy Selection to Clipboard',
+        mnemonic: -1,
+        isEnabled: () => {
+          return this._isSelectionSelected();
```
Comment (martinRenou) [inline js/feathergrid.ts:998]:
```suggestion
          return this.grid.selectionModel && !this.grid.selectionModel.isEmpty;
```

File: js/feathergrid.ts [hunk 3]
```diff
@@ -994,6 +989,56 @@ export class FeatherGrid extends Widget {
...
+        execute: () => {
+          this.grid.copyToClipboard();
+        },
+      },
+    );
+    commands.addCommand(FeatherGridContextMenu.CommandID.SaveSelectionAsCsv, {
+      label: 'Download Selection as CSV',
+      mnemonic: -1,
+      isEnabled: () => {
+        return this._isSelectionSelected();
```
Comment (martinRenou) [inline js/feathergrid.ts:1009]:
```suggestion
        return this.grid.selectionModel && !this.grid.selectionModel.isEmpty;
```

File: js/feathergrid.ts [hunk 4]
```diff
@@ -885,6 +871,15 @@ export class FeatherGrid extends Widget {
...
 
+  _isSelectionSelected(): boolean {
+    if (this.grid.selectionModel && !this.grid.selectionModel.isEmpty) {
+      return true;
+    } else {
+      return false;
+    }
+    // return this.grid.selectionModel && this.grid.selectionModel.isEmpty;
+  }
+
```
Comment (martinRenou) [inline js/feathergrid.ts:882]: I'd remove this and just use the expression in the `isEnabled` option

```suggestion
```

**Review by martinRenou — CHANGES_REQUESTED** [review]
Thanks a lot!

I have a couple of small suggestions

File: js/feathergrid.ts (hunk 4 above)
Comment (gjmooney) [reply js/feathergrid.ts:882]: `isEnabled` requires a true/false value, and `this.grid.selectionModel` can potentially be `null` , I think it should work if I use `!!this.grid.selectionModel && !this.grid.selectionModel.isEmpty` if that's preferable.

Comment (martinRenou) [reply js/feathergrid.ts:882]: or `this.grid.selectionModel == null || this.grid.selectionModel.isEmpty`

**Comment by martinRenou** [comment]
Bot update galata references

**Comment by martinRenou** [comment]
CI failure is fixed in https://github.com/bloomberg/ipydatagrid/pull/482

Let's merge and iterate, thanks!

**Review by martinRenou — APPROVED** [review]

**Pull request merged**